desc/equilibrium/coords.py

```diff
@@ -181,6 +161,66 @@ def _distance_body(i, idx):
     return out
 
 
+def _initial_guess_heuristic(yk, coords, inbasis, eq, profiles):
+    # some qtys have obvious initial guess based on coords
+    # commonly, the desired coordinates are something like (radial, poloidal, toroidal)
+    radialish = {"rho", "psi"}
+    poloidalish = {"theta", "theta_PEST", "alpha"}
+    toroidalish = {"zeta", "phi"}
+
+    radial = list(set(inbasis).intersection(radialish))
+    poloidal = list(set(inbasis).intersection(poloidalish))
+    toroidal = list(set(inbasis).intersection(toroidalish))
+    if len(radial) != 1 or len(poloidal) != 1 or len(toroidal) != 1:
+        # no heuristics for other cases
+        return yk
+
+    rho = theta = zeta = None
+
+    radial = radial[0]
+    poloidal = poloidal[0]
+    toroidal = toroidal[0]
+    if radial == "rho":
+        rho = coords[:, inbasis.index("rho")]
+    elif radial == "psi":
+        rho = jnp.sqrt(coords[:, inbasis.index("psi")] / eq.Psi)
+
+    # omega usually small (zero for now)
+    zeta = coords[:, inbasis.index(toroidal)]
+
+    if poloidal == "theta" or poloidal == "theta_PEST":  # lambda usually small
+        theta = coords[:, inbasis.index(poloidal)]
+    elif poloidal == "alpha":
+        alpha = coords[:, inbasis.index("alpha")]
+        iota = profiles["iota"](rho)
+        theta = (alpha + iota * zeta) % (2 * jnp.pi)
```

Comment:
When we're setting straight field line theta as the initial guess for desc theta,
I think this formula is only accurate if `zeta` is the toroidal coordinate in DESC coords.
If it is field line `zeta` then $(\rho, \alpha, \zeta) \neq (\rho, \alpha, \zeta + 2 \pi)$.
Why doesn't it make more sense to not modulo by 2pi for the initial guess?